source/module_io/write_dipole.cpp

```diff
@@ -43,12 +43,12 @@ void ModuleIO::write_dipole(const double* rho_save,
         {
             for (int i = 0; i < rhopw->nx; i++)
             {
-                dipole_elec_x += rho_save[i * rhopw->ny * rhopw->nz + j * rhopw->nz + k] * i
-                                 * GlobalC::ucell.lat0 * 0.529177 / rhopw->nx;
-                dipole_elec_y += rho_save[i * rhopw->ny * rhopw->nz + j * rhopw->nz + k] * j
-                                 * GlobalC::ucell.lat0 * 0.529177 / rhopw->ny;
-                dipole_elec_z += rho_save[i * rhopw->ny * rhopw->nz + j * rhopw->nz + k] * k
-                                 * GlobalC::ucell.lat0 * 0.529177 / rhopw->nz;
```

Comment:
i * rhopw->ny * rhopw->nz + j * rhopw->nz + k can be calculated once and saved in a variable